engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -110,7 +111,9 @@ class NioFlow(parallelism: Int,
     hash.file match {
       case gcsPath: GcsPath => IO { gcsPath.cloudStorage.get(gcsPath.blob).getCrc32c }
       case drsPath: DrsPath => getFileHashForDrsPath(drsPath)
-      case path => IO.fromEither(
+      case s3Path: S3Path => IO { s3Path.eTag }
+      case path =>
```

Comment:
FWIW all that happens now is it throws an error and defaults to running the WF so I'm not sure that'd be gaining anything in a practical sense. OTOH I **am** the person who repeatedly made the claim that type safety was its own reward (ignoring that `asInstanceOf` in this PR!)

Up to you if you think it's worth having

Reply:
Yeah, not a sealed trait, so no compile time errors. But it could be a clearer runtime error by doing Chris' suggestion.

![alfre_and_cromwell_small](https://user-images.githubusercontent.com/791985/55512286-bc5ad880-5630-11e9-834a-0041d102b58f.png)

Someday...

Reply:
Fair 'nuff.